### DISCUSSION TITLE
Is required keyword reserved for requests only?

### DISCUSSION THREAD

**Discussion by Zahak** [OAI/OpenAPI-Specification#3778] [Q&A] [answered]
Hi,

we have been using required keyword both for request and response objects. Our idea was to show which fields in the response that will always be there, and which might be there. 

Now there is debate if that is correct, as some say required keyword should only be used for requests.

so for starters, is that correct, that it shouldn’t be used for responses? 

If so, how should fields that will always be in response and those that might be there be differentiated?

also, if anyone could post some links to some info about the subject that would be great. 

cheers!

**Comment by handrews** [comment]
@Zahak If you mean a `required` field for the whole response, the specification states where `required` can be used, and it is **_not_** present in the Response Object.  We do already have two issues tracking questions around optional response content:

* #2236 
* #3536 

If you mean individual fields, `required` works in the Schema Object the same way regardless of whether it is a request or response.

**Reply by Zahak** [reply]
Thanks for quick reply. I’m thinking about individual fields. Let’s say the response would have fields phone number and work number, where phone number always is returned hence required, while work number is optional and may be returned if exists. 

So then that logic would be fine, right?

**Reply by handrews** [reply, marked as answer]
Yes, you would put `phone_number` (or however you spell the field) in the schema's `required` keyword, but leave the `work_number` field out.  Which is exactly what you would do for requests.  The Schema Object doesn't "know" whether it's a request or response or parameter or header or whatever - it's just a structural description.

**Comment by handrews** [comment]
Answer accepted, closing.
